plugins/salesforce/src/org/pentaho/di/trans/steps/salesforce/SalesforceConnection.java

```diff
@@ -273,6 +273,9 @@ public void setPassword( String value ) {
   public void connect() throws KettleException {
     ConnectorConfig config = new ConnectorConfig();
     config.setAuthEndpoint( getURL() );
+    config.setServiceEndpoint( getURL() );
+    config.setUsername( getUsername() );
+    config.setPassword( Encr.decryptPasswordOptionallyEncrypted( getPassword() ) );
```

Comment:
It appears to have been happening as I wasn't returning the ConnectorConfig object, due to the series of mocks.  It's been updated to pass back a mock ConnectorConfig, as we just care about the original that was passed in, that it had the correct credentials.